### ISSUE TITLE
Investigate the size of memory leaks

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#10749]
To unblock #5954 we need to figure out what issues we have currently.

*   Confirmation (that the bug occurs without the Inspector, make sure to call garbage collector manually before the last snapshot)
*   Analysis of memory profiler logs to figure out what kind of retainer chains are there

**Comment by oleq** [comment]
TL;DR: Turns out, the editor does not leak much in successive init&destroy tests (core features). It's the MathType+ChemType that causes issues.

I ran a series of 10 `init()`/`destroy()` of the editor, each destroy followed by GC on demand (then memory snapshot). There are a few things retained between each run, the memory usage rises and drops. However, when I enabled the `MathType` plugin I noticed that the memory usage grew tenfold

<img width="246" alt="Screenshot 2021-11-04 at 09 50 46" src="https://user-images.githubusercontent.com/1099479/140307182-a082470b-0391-4720-8de6-76f2f7aacb94.png">

A quick look at the snapshot revealed that the `MathType` integration retains many editing view elements (despite no math type widgets loaded in the data). Looks like this has something to do with the way the integration attaches event listeners for clicking, etc.; this needs further investigation

**Comment by oleq** [comment]
FYI: There is a PR related to the research (new manual test + safeguarded some bits of UI) #10803.
